torch/_inductor/pattern_matcher.py

```diff
@@ -631,11 +627,6 @@ def run_node(self, node) -> Any:
                     target = node.target
                     args, kwargs = self.fetch_args_kwargs_from_env(node)
                     result = graph.call_function(target, args, kwargs)
-                    # Retain the meta tags from the first node in the match.
-                    # This is useful for retaining tags like recompute.
-                    for key in first_node.meta.keys():
-                        if key in preserve_meta_keys:
-                            result.meta[key] = first_node.meta[key]
```

Comment:
I guess just propagating only the first_node was insufficient?

Reply:
Yeah. Actually it was incorrect. For training graph, you could have multiple outputs in the replacement graph, some for forward outputs and some for out_grads. Previous logic was just using first_node (which is first forward output) to tag all the nodes in the replacement graph.